app/workers/subject_removal_worker.rb

```diff
@@ -0,0 +1,12 @@
+require 'subjects/remover'
+
+class SubjectRemovalWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :data_medium
+
+  def perform(subject_id)
+    Subjects::Remover.new(subject_id).cleanup
+  rescue Subjects::Remover::NonOrphan
```

Comment:
This rescue ensures says not an orphaned subject so move along and don't keep trying via sidekiq.